From 84f23f58fa06867e6f22523a6df4a84e49cdf515 Mon Sep 17 00:00:00 2001
From: Sergey Temerkhanov <s.temerkhanov@gmail.com>
Date: Sat, 20 Feb 2016 13:39:55 -0800
Subject: [PATCH 0381/1239] PCIe: Skip PEM enumeration

Don't enumerate PEMs and devices under them
since it might lead to very long delays

Signed-off-by: Sergey Temerkhanov <s.temerkhanov@gmail.com>
---
 board/cavium/thunderx/pcie.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/board/cavium/thunderx/pcie.c b/board/cavium/thunderx/pcie.c
index c897a3f472..f5de550406 100644
--- a/board/cavium/thunderx/pcie.c
+++ b/board/cavium/thunderx/pcie.c
@@ -459,7 +459,12 @@ int pci_config_fixed(struct pci_controller *hose, pci_dev_t dev)
 
 int pci_skip_dev(struct pci_controller *hose, pci_dev_t dev)
 {
-        return 0;
+	u16 vendor, device;
+
+	pci_hose_read_config_word(hose, dev, PCI_DEVICE_ID, &device);
+	pci_hose_read_config_word(hose, dev, PCI_VENDOR_ID, &vendor);
+
+	return (vendor == PCI_VENDOR_CAVIUM && device == PCI_DEV_CAVIUM_RC);
 }
 
 int pci_print_dev(struct pci_controller *hose, pci_dev_t dev)
-- 
2.29.0

